### PR TITLE
Deploy to staging via Nix

### PR DESCRIPTION
**Pull request by pingiun**
### Summary
Deploy to staging via Nix. I think this PR depends on a better fix to #992 because Nix doesn't allow internet access when building. It seems that networkless building still fails.

### How to test
If you are a trusted user on the staging machine, you can run
```bash
scripts/send-conf-and-switch.sh
```
to apply the latest configuration to the staging NixOS server

### CODE REVIEW

File: nix/configuration.nix [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+      };
+    };
+
+    # These configuration options are defined and implemented by the timed-commands.nix file
+    concrexit-timers = {
+      send_scheduled_messages = {
+        every = 60 * 5;
+        description = "Send scheduled push notifications";
+      };
+      sendplannednewsletters.every = 60 * 60;
```
Comment (Legosteen11) [inline nix/configuration.nix:211]: I think we should add some more descriptions

File: nix/configuration.nix [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    environment.systemPackages = [ pkgs.concrexit-manage ];
+    networking.hostName = "staging";
+
+    security.acme.email = "www@thalia.nu";
+    security.acme.acceptTerms = true;
+
+    nix = {
+      gc.automatic = true;
+      # This is required to be able to use nix-copy-closure
+      trustedUsers = [ "root" "deploy" "jelle" ];
```
Comment (Legosteen11) [inline nix/configuration.nix:121]: I will add myself as well :wink:

File: nix/configuration.nix [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    services = {
+      nginx = {
+        enable = true;
+
+        recommendedGzipSettings = true;
+        recommendedOptimisation = true;
+        recommendedTlsSettings = true;
+
+        # Reloads NGINX instead of restarting it whenever possible
+        enableReload = true;
```
Comment (Legosteen11) [inline nix/configuration.nix:232]: Let's just always restart nginx

File: nix/configuration.nix [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+              extraConfig = securityHeaders;
+            };
+          in
+          {
+            "${cfg.domain}" = {
+              # Request a certificate from letsencrypt
+              enableACME = cfg.ssl;
+              # Enable redirects to https
+              forceSSL = cfg.ssl;
+              default = !cfg.ssl;
```
Comment (Legosteen11) [inline nix/configuration.nix:250]: ???

File: nix/configuration.nix [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+              '';
+            };
+            "www.${cfg.domain}" = {
+              enableACME = cfg.ssl;
+              locations."/".return = "301 https://${cfg.domain}$request_uri";
+              extraConfig = securityHeaders;
+            };
+            "pizza.${cfg.domain}" = pizzaConfig;
+            "pasta.${cfg.domain}" = pizzaConfig;
+            "xn--vi8h.${cfg.domain}" = pizzaConfig;
```
Comment (Legosteen11) [inline nix/configuration.nix:278]: Add pasta emoji (xn--3i8h, 🍝)

File: resources/maintenance.html [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	.st0{fill:#E62272;}
+</style>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+</head>
+<body>
+<article>
+<svg version="1.1" id="image" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="-233 408.5 374 375" style="enable-background:new -233 408.5 374 375;" xml:space="preserve">
+<g>
+	<path id="hoekje_onder_3_" class="st0" d="M25.3,712.7c-4.4,0,89.8,10.9,91.2,10.9l-51.6,40.1L25.3,712.7L25.3,712.7z"/>
```
Comment (Legosteen11) [inline resources/maintenance.html:22]: Panic!!! We have introduced more Dutch language on the concrexit site 😱😱😱🚨🚨🚨

**Review by Legosteen11 — CHANGES_REQUESTED** [review]
Looks good, I will improve it a bit

File: resources/maintenance.html (hunk 6 above)
Comment (pingiun) [reply resources/maintenance.html:22]: Lol

File: nix/configuration.nix (hunk 1 above)
Comment (Legosteen11) [reply nix/configuration.nix:211]: I should add an issue to ask someone to explain what these management commands do

File: nix/configuration.nix [hunk 7]
```diff
@@ -76,6 +76,11 @@ in
       default = true;
     };
 
+    concrexit.rejectUnknownHost = mkOption {
+      type = types.bool;
+      default = cfg.ssl;
```
Comment (pingiun) [inline nix/configuration.nix:81]: If this is the default, the option should be named `allowUnknownHost` or the logic should be reversed. Just run through it and you'll see what I mean

File: nix/configuration.nix (hunk 1 above)
Comment (pingiun) [reply nix/configuration.nix]: Yes good idea

File: nix/configuration.nix (hunk 7 above)
Comment (pingiun) [reply nix/configuration.nix:81]: Also I recommend to add a description to this option

**Review by Legosteen11 — APPROVED** [review]

**Pull request merged**